arch/x86/core/pcie.c

```diff
@@ -197,7 +203,15 @@ uint32_t pcie_msi_map(unsigned int irq,
 #endif
 #endif
 	{
-		map = 0xFEE00000U; /* standard delivery to BSP local APIC */
+		uint32_t dest_id;
+
+		dest_id = z_x86_cpuid_get_current_physical_apic_id() <<
+			MSI_MAP_DESTINATION_ID_SHIFT;
+
+		/* Directing to current physical CPU (may not be BSP)
+		 * Destination ID - RH 1 - DM 0
+		 */
+		map = 0xFEE00000U | dest_id | MSI_RH;
```

Comment:
Could be useful to keep the comment /* standard delivery to BSP local APIC */ associated with 0xFEE00000U. e.g.:

```
map = 0xFEE00000U; /* standard delivery to BSP local APIC */
/* Directing to current physical CPU (may not be BSP)
  * Destination ID - RH 1 - DM 0
  */
map |= dest_id | MSI_RH;
```
or something similar

Reply:
No the patch is exactly removing that. And here we are on a backport so it's not going to diverge more than the original one that landed in 3.0.

0xFEE00000U is the local apic address. It's also the bsp local apic because destination address for BSP is 0.

Reply:
Thanks for the feedback.
I missed the part this was backported. Issue should be addressed in the upstream branch. Avoid magic numbers and write self describing code. :)